src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -118,14 +155,64 @@ protected void processAccountingDocumentFromXml(String fileName, CreateAccountin
             removeDoneFileQuietly(fileName);
             createAndEmailReport(reportItem);
         }
-        if (reportItem.isNonBusinessRuleFailure()) {
+        if (reportItem.isDuplicateFile() && shouldExcludePreviouslyProcessedFiles()) {
+            logReport.getExcludedDuplicateFiles().add(fileName);
+            logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
+        } else if (reportItem.isNonBusinessRuleFailure()) {
             logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
         } else {
             logReport.getFilesSuccessfullyProcessed().add(fileName);
         }
         LOG.info("processAccountingDocumentFromXml: Value of reportItem.isNonBusinessRuleFailure just prior to method return =" + reportItem.isNonBusinessRuleFailure());
     }
-    
+
+    protected CreateAccountingDocumentFileEntry findExistingFileEntry(String fileName) {
+        String cleanedFileName = getCleanedFileName(fileName);
+        return businessObjectService.findBySinglePrimaryKey(CreateAccountingDocumentFileEntry.class, cleanedFileName);
+    }
+
+    protected String getCleanedFileName(String fileName) {
+        String cleanedFileName = fileName;
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.BACKSLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.BACKSLASH);
+        }
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.SLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.SLASH);
+        }
+        cleanedFileName = StringUtils.lowerCase(cleanedFileName, Locale.US);
+        return cleanedFileName;
+    }
+
+    private boolean shouldExcludePreviouslyProcessedFiles() {
+        return parameterService.getParameterValueAsBoolean(KFSConstants.CoreModuleNamespaces.FINANCIAL, 
+                CuFPParameterConstants.CreateAccountingDocumentService.CREATE_ACCOUNTING_DOCUMENT_SERVICE_COMPONENT_NAME, 
+                CuFPParameterConstants.CreateAccountingDocumentService.DUPLICATE_FILE_CHECK_IND);
+    }
+
+    private AccountingXmlDocumentListWrapper unmarshalAccountingDocumentFile(String fileName) {
+        byte[] fileData = LoadFileUtils.safelyLoadFileBytes(fileName);
+        return (AccountingXmlDocumentListWrapper) batchInputFileService.parse(
+                accountingDocumentBatchInputFileType, fileData);
+    }
+
+    private void createOrUpdateFileEntry(AccountingXmlDocumentListWrapper accountingXmlDocuments,

```

Comment:
I find this convention a bit confusing, and perhaps error prone with potential future updates.  If an XML was previously loaded, we don't want to update the values for existingEntry.  This naming convention makes it sound like that is possible.  Can this be refactored to help with that?

Reply:
I can modify this to only support creating entries and not updating them. I wasn't sure if we needed to allow for updating existing entries when duplication is permitted, but if you're fine with the file entries only reflecting the first processed file of a given name (as opposed to the latest file), then I'm fine with making this a create-only method.

Reply:
I don't think we want to adjust previously saved records, they should be a historic record of a processed file